src/main/java/com/iota/iri/storage/rocksDB/RocksDBPPPImpl.java

```diff
@@ -0,0 +1,596 @@
+     * @param key index key
+     * @param indexValue index value
+     * @throws Exception
+     */
+    @VisibleForTesting
+    void addToIndex(WriteBatch writeBatch, ColumnFamilyHandle column, Indexable key, Indexable indexValue)
+            throws Exception {
+        byte[] indexBytes = indexValue.bytes();
+
+        byte[] dbResult = db.get(column, key.bytes());
+
+        if (dbResult != null && Bytes.indexOf(dbResult, indexBytes) != -1) {// not found
+
+            ByteBuffer buffer = ByteBuffer
+                    .allocate(dbResult.length + indexBytes.length + (dbResult.length > 0 ? 1 : 0));// +1 delimiter
+                                                                                                   // length
+            buffer.put(dbResult);
+            if (dbResult.length > 0) {
+                // Means we add on the end of the current stream.
+                // To be compatible with Hashes.read(byte[])
+                buffer.put(delimiter);
+            }
+            buffer.put(indexBytes);
+            writeBatch.put(column, key.bytes(), buffer.array());
+        }
+    }
+
+    /**
+     * @see PermanentPersistenceProvider#getTransaction(Hash)
+     */
+    @Override
+    public TransactionViewModel getTransaction(Hash index) throws Exception {
+        if (index == null) {
+            return null;
+        }
+        byte[] dbResult = db.get(columnMap.get(TRANSACTION_COLUMN), index.bytes());
+        if (dbResult != null && dbResult.length > 0) {
+            return new TransactionViewModel(TransactionViewModel.trits(dbResult), Hash.NULL_HASH);
+        }
+        return null;
+    }
+    /**
+     * @see PermanentPersistenceProvider#findAddress(Hash)
+     */
+    @Override
+    public Hashes findAddress(Hash index) throws Exception {
+        return getIndex(columnMap.get(ADDRESS_INDEX), index);
+    }
+    /**
+     * @see PermanentPersistenceProvider#findBundle(Hash)
+     */
+    @Override
+    public Hashes findBundle(Hash index) throws Exception {
+        return getIndex(columnMap.get(BUNDLE_INDEX), index);
+    }
+    /**
+     * @see PermanentPersistenceProvider#findTag(Hash)
+     */
+    @Override
+    public Hashes findTag(Hash index) throws Exception {
+        return getIndex(columnMap.get(TAG_INDEX), index);
+    }
+    /**
+     * @see PermanentPersistenceProvider#findApprovee(Hash)
+     */
+    @Override
+    public Hashes findApprovee(Hash index) throws Exception {
+        return getIndex(columnMap.get(APPROVEE_INDEX), index);
+    }
+
+    private Hashes getIndex(ColumnFamilyHandle column, Indexable index) throws Exception {
+        if (index == null) {
+            return new Hashes();
+        }
+        Hashes found = new Hashes();
+        byte[] result = db.get(column, index.bytes());
+        if (result != null) {
+            found.read(result);
+        }
+        return found;
+    }
+
+    private void initDB(String path, String logPath) throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'initDB(String, String)' has a cyclomatic complexity of 10.](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4243662)